File: api/src/main/java/com/FireSale/api/dto/WebsocketAuctionMessage.java
package com.FireSale.api.dto;
import lombok.AllArgsConstructor;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@AllArgsConstructor
public class WebsocketAuctionMessage<T>{
    private ResponseType responseType;
    private T data;
    private long userId;
    private LocalDateTime messageTime;
}

File: client/FireSale/src/app/shared/models/webSocketAuctionMessage.ts
export interface WebSocketAuctionMessage<T> {
  responseType: AuctionMessageResponseType;
  data: T;
  userId: number;
  messageTime: Date;
}
export enum AuctionMessageResponseType {
  BID_PLACED = 'BID_PLACED',
  UPDATED = 'UPDATED'
}
